WebHostLib/generate.py

```diff
@@ -76,6 +76,10 @@ def format_exception(e: BaseException) -> str:
     return f"{e.__class__.__name__}: {e}"
 
 
+class GenTaskGenerationError(BaseException):
```

Comment:
Is there a reason to inherit from `BaseException` here instead of `Exception`?

Reply:
This PR has changed again now, so I'm resolving this